crate/server/src/core/operations/export_utils.rs

```diff
@@ -653,3 +659,82 @@ async fn post_process_pkcs12_for_private_key(
     };
     Ok(())
 }
+
+async fn post_process_pkcs7(
+    kms: &KMS,
+    operation_type: ObjectOperationType,
+    user: &str,
+    params: Option<&ExtraDatabaseParams>,
+    owm: ObjectWithMetadata,
+) -> Result<ObjectWithMetadata, KmsError> {
+    // convert the cert to openssl
+    let certificate = kmip_certificate_to_openssl(&owm.object)
+        .context("export: unable to parse the certificate to openssl")?;
+
+    let mut cert_owm = owm.clone();
+
+    let leaf_cert = certificate.clone();
+    let public_key_id = cert_owm
+        .attributes
+        .get_link(LinkType::PublicKeyLink)
+        .ok_or_else(|| {
+            KmsError::InvalidRequest("No Public Key found in the leaf certificate".to_string())
```

Comment:
The error type `InvalidRequest` is not obvious for me. Shoud it be a `KmipError` instead? (since this KMIP object is invalid)